File: frontend/src/types/prediction.ts
/** Matches backend/models/prediction.py */

export interface CustomerRiskPredictionRow {
  customer_id: string
  risk_probability: number | null
  predicted_class: number | null
  risk_level: string | null
  model_version: string
  generated_at: string
}

export interface ProjectRiskPredictionRow {
  project_id: string
  risk_probability: number | null
  predicted_risk_class: number | null
  risk_level: string | null
  model_version: string
  generated_at: string
}

export interface PaymentDelayPredictionRow {
  invoice_id: string
  customer_id: string
  delay_probability: number | null
  predicted_delay: number | null
  risk_level: string | null
  model_version: string
  generated_at: string
}

File: backend/models/prediction.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CustomerRiskPredictionRow(BaseModel):
    customer_id: str
    risk_probability: Optional[float] = None
    predicted_class: Optional[int] = None
    risk_level: Optional[str] = None
    model_version: str
    generated_at: datetime


class ProjectRiskPredictionRow(BaseModel):
    project_id: str
    risk_probability: Optional[float] = None
    predicted_risk_class: Optional[int] = None
    risk_level: Optional[str] = None
    model_version: str
    generated_at: datetime


class PaymentDelayPredictionRow(BaseModel):
    invoice_id: str
    customer_id: str
    delay_probability: Optional[float] = None
    predicted_delay: Optional[int] = None
    risk_level: Optional[str] = None
    model_version: str
    generated_at: datetime
